Sage/carbon
# fix(use-scroll-block): remove right padding added to the body

### Summary

Once the `useScrollBlock` is called, the `clientWidth` of the document is retrieved to calculate the scroll width in a memoized value. If the `clientWidth` changes then the value should be re-calculated.

fixes #5220

### Proposed behaviour

the `clientWidth` is added as a dependency of the `useMemo` hook, so every time the `useScrollBlock` is called, the value is re-calculated if changed.

### Current behaviour

Currently, as the `useMemo` has an empty array of dependencies, it's only called once. If after that, the `clientWidth` changes, for instance, due to router navigation, the value will be out of date.

### Checklist


#### QA


### Additional context

N/A

### Testing instructions

For more info about the broken behaviour, see #5220  
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.
